File: Clases 2018/Clases-C#/ClasePractica12_2018 (transacciones con data manager)/C#_TPS_BugTracker/TPS_BugTracker/DataLayer/DAOs/PerfilDao.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TPS_BugTracker.BusinessLayer.Entities;

namespace TPS_BugTracker.DataLayer.DAOs
{
    class PerfilDao
    {
        public List<Perfil> getAll()
        {
            List<Perfil> lst = new List<Perfil>();
            DataManager dm = new DataManager();
            try
            {
                dm.Open();
                foreach (DataRow row in dm.ConsultaSQL("Select * from Perfiles").Rows)
                    lst.Add(map(row));

            }
            catch (Exception ex)
            {
                dm.Rollback();
            }
            finally
            {
                // Cierra la conexión 
                dm.Close();
            }

            return lst;
        }

        private Perfil map(DataRow row)
        {
            Perfil Perfil = new Perfil();
            Perfil.id_perfil = Convert.ToInt32(row["id_perfil"].ToString());
            Perfil.nombre = row["n_perfil"].ToString();
            return Perfil;
        }
    }
}

File: Clases 2018/Clases-C#/Simulacro_parcial_2_2018 (creacion de bug e historico en transaccion)/VB_TPS_BugTracker/TPS_BugTracker/DataLayer/DAOs/PerfilDao.vb
Class PerfilDao
    Public Function getAll() As List(Of Perfil)
        Dim lst As List(Of Perfil) = New List(Of Perfil)()
        Dim dm As DataManager = New DataManager()

        Try
            dm.Open()

            For Each row As DataRow In dm.ConsultaSQL("Select * from Perfiles").Rows
                lst.Add(map(row))
            Next

        Catch ex As Exception
            dm.Rollback()
        Finally
            dm.Close()
        End Try

        Return lst
    End Function

    Private Function map(ByVal row As DataRow) As Perfil
        Dim Perfil As Perfil = New Perfil()
        Perfil.id_perfil = Convert.ToInt32(row("id_perfil").ToString())
        Perfil.nombre = row("n_perfil").ToString()
        Return Perfil
    End Function
End Class